### ISSUE TITLE
About WAE Adversarial Block Layer

### ISSUE THREAD

**Issue by zhaihongjia** [ymcidence/TBH#3]
The model layers are shown in the following:

        self.encoder = encodec.Encoder(middle_dim, bbn_dim, cbn_dim)
        self.decoder = encodec.Decoder(middle_dim, self.feat_dim)
        self.tbn = twin_bottleneck.TwinBottleneck(bbn_dim, cbn_dim)
        self.dis_1 = tf.keras.layers.Dense(1, activation='sigmoid')
        self.dis_2 = tf.keras.layers.Dense(1, activation='sigmoid')

 The WAE Adversarial Block Layer: cbn->1024->1 & bbn->1024->1
 Maybe you lack a FC layer with 1024 nodes in tbh.py?

**Comment by ymcidence** [comment]
> The model layers are shown in the following:
> 
> ```
>     self.encoder = encodec.Encoder(middle_dim, bbn_dim, cbn_dim)
>     self.decoder = encodec.Decoder(middle_dim, self.feat_dim)
>     self.tbn = twin_bottleneck.TwinBottleneck(bbn_dim, cbn_dim)
>     self.dis_1 = tf.keras.layers.Dense(1, activation='sigmoid')
>     self.dis_2 = tf.keras.layers.Dense(1, activation='sigmoid')
> ```
> 
> The WAE Adversarial Block Layer: cbn->1024->1 & bbn->1024->1
> Maybe you lack a FC layer with 1024 nodes in tbh.py?

During re-implementation, we find that simply removing one layer does not harm the overall performance as adversarial regularization is just an auxiliary module in our model. Hence, we drop one layer. This will be reflected in our journal extension. If  one needs to reproduce the exact model described in our CVPR paper, simply adding one more layer will do.
